mriqc/interfaces/functional.py

```diff
@@ -309,6 +311,189 @@ def _run_interface(self, runtime):
         return runtime
 
 
+class GatherTimeseriesInputSpec(TraitedSpec):
+    dvars = File(exists=True, mandatory=True, desc='file containing DVARS')
+    fd = File(exists=True, mandatory=True, desc='input framewise displacement')
+    mpars = File(exists=True, mandatory=True, desc='input motion parameters')
+    mpars_source = traits.Enum(
+        "FSL",
+        "AFNI",
+        "SPM",
+        "FSFAST",
+        "NIPY",
+        desc="Source of movement parameters",
+        mandatory=True,
+    )
+    outliers = File(
+        exists=True,
+        mandatory=True,
+        desc="input file containing timeseries of AFNI's outlier count")
+    quality = File(
+        exists=True,
+        mandatory=True,
+        desc="input file containing AFNI's Quality Index")
+
+
+class GatherTimeseriesOutputSpec(TraitedSpec):
+    timeseries_file = File(
+        exists=True,
+        desc='output confounds file',
+        resolve=True,
+        extensions="tsv")
+    timeseries_metadata = traits.Dict
+
+
+class GatherTimeseries(SimpleInterface):
+    """
+    Gather quality metrics that are timeseries into one TSV file
+
+    """
+    input_spec = GatherTimeseriesInputSpec
+    output_spec = GatherTimeseriesOutputSpec
+
+    def _run_interface(self, runtime):
+
+        # motion parameters
+        mpars = np.apply_along_axis(
+            func1d=normalize_mc_params,
+            axis=1,
+            arr=np.loadtxt(self.inputs.mpars),  # mpars is N_t x 6
+            source=self.inputs.mpars_source,
+        )
+        timeseries = pd.DataFrame(
+            mpars,
+            columns=[
+                "trans_x",
+                "trans_y",
+                "trans_z",
+                "rot_x",
+                "rot_y",
+                "rot_z"
+            ])
+
+        # DVARS
+        dvars = pd.read_csv(
+            self.inputs.dvars,
+            delim_whitespace=True,
+            skiprows=1,  # column names have spaces
+            header=None,
+            names=["dvars_std", "dvars_nstd", "dvars_vstd"])
+        dvars.index = pd.RangeIndex(1, timeseries.index.max() + 1)
+
+        # FD
+        fd = pd.read_csv(
+            self.inputs.fd,
+            delim_whitespace=True,
+            header=0,
+            names=["framewise_displacement"])
+        fd.index = pd.RangeIndex(1, timeseries.index.max() + 1)
+
+        # AQI
+        aqi = pd.read_csv(
+            self.inputs.quality,
+            delim_whitespace=True,
+            header=None,
+            names=["aqi"])
+
+        # Outliers
+        aor = pd.read_csv(
+            self.inputs.outliers,
+            delim_whitespace=True,
+            header=None,
+            names=["aor"])
+
+        timeseries = pd.concat((timeseries, dvars, fd, aqi, aor), axis=1)
+
+        timeseries_file = "timeseries.tsv"
```

Comment:
I've generally preferred using `runtime.cwd` and returning resolved paths. Although the CWD should match whenever you have the `runtime` object, you're less likely to find some weird edge case where something needs to resolve the path when the process CWD has changed.

```suggestion
        timeseries_file = op.join(runtime.cwd, "timeseries.tsv")
```